### PR TITLE
SONAR-8574 Replace usage of api/resources/index by api/measures/component in ITs

### PR DESCRIPTION
**Pull request by julienlancelot**
…nent in ITs

### CODE REVIEW

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java [hunk 1]
```diff
@@ -40,40 +38,14 @@
...
-  private static List<String> ruleActivations(Map actives, Map<String, String> jsProfileNameById, String ruleKey) {
-    List<String> profileNames = new ArrayList<>();
-    List<Map> activations = (List) actives.get(ruleKey);
-    for (Map activation : activations) {
-      String profileId = (String) activation.get("qProfile");
-      profileNames.add(jsProfileNameById.get(profileId));
-    }
-    return profileNames;
+    assertThat(actives.get("common-js:InsufficientCommentDensity").getActiveListList()).isEmpty();
+    assertThat(actives.get("common-js:DuplicatedBlocks").getActiveListList()).hasSize(1);
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java]: @julienlancelot I hardly understand what is happening here, but can we add assert on profile? (unless we already doing that)

Comment (julienlancelot) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java]: @vilchik-elena Here the goal is to check that the rule "common-js:DuplicatedBlocks" is activated on one profile.
It's the using the WS api/rules/search that allow to returned the list of profiles where a rule is activated (using the parameter fields to "actives").
Does it answer your question ?

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java [hunk 2]
```diff
@@ -51,92 +48,90 @@ public static void prepare() {
...
-    assertThat(getFileMeasure("functions").getIntValue()).isEqualTo(5);
-    assertThat(getFileMeasure("files").getIntValue()).isEqualTo(1);
+    assertThat(getFileMeasureAsDouble("functions")).isEqualTo(5);
+    assertThat(getFileMeasureAsDouble("files")).isEqualTo(1);
     // Documentation
-    assertThat(getFileMeasure("comment_lines").getIntValue()).isEqualTo(1);
-    assertThat(getFileMeasure("commented_out_code_lines")).isNull();
-    assertThat(getFileMeasure("comment_lines_density").getValue()).isEqualTo(4.8);
+    assertThat(getFileMeasureAsDouble("comment_lines")).isEqualTo(1);
+    assertThat(getFileMeasureAsDouble("commented_out_code_lines")).isZero();
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java]: @julienlancelot Why this change?

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java [hunk 3]
```diff
@@ -51,92 +48,90 @@ public static void prepare() {
...
+    assertThat(getFileMeasureAsDouble("comment_lines_density")).isEqualTo(4.8);
     // Complexity
-    assertThat(getFileMeasure("file_complexity_distribution")).isNull();
+    assertThat(getFileMeasureAsDouble("file_complexity_distribution")).isNull();
     // Duplication
-    assertThat(getFileMeasure("duplicated_lines")).isNull();
-    assertThat(getFileMeasure("duplicated_blocks")).isNull();
-    assertThat(getFileMeasure("duplicated_files")).isNull();
-    assertThat(getFileMeasure("duplicated_lines_density")).isNull();
+    assertThat(getFileMeasureAsDouble("duplicated_lines")).isZero();
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java]: @julienlancelot And here as well null->zero

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java (hunk 1 above)
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java]: @julienlancelot Ok, then with new implementation we asserting that rule is activated at ONE profile (whatever it is), but previously we were asserting that it's activated on "Sonar way" profile. 
So how can we check on which profile rule is activated?
Thanks!

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java (hunk 2 above)
Comment (julienlancelot) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java]: Because 'commented_out_code_lines' is using best values => When there's no measure, it's returning the best value, which is 0 for this measure.
Old WS api/resources/index was not handling best values, by new one is handling them.

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java (hunk 3 above)
Comment (julienlancelot) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/MetricsTest.java]: Same answer, duplicated_lines is using best values (1).

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java (hunk 1 above)
Comment (julienlancelot) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java]: Good catch ! For sure it's possible to check that the rule is activated on the Sonar way profile.
I'm fixing it.

Comment (julienlancelot) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/DefaultProfileTest.java]: @vilchik-elena It should be better now

**Comment by vilchik-elena** [comment]
LGTM, Thanks!
